lib/gtfs/data.ex

```diff
@@ -95,6 +105,19 @@ defmodule Gtfs.Data do
     |> Helpers.merge_lists()
   end
 
+  @spec directions_by_route_id(binary()) :: directions_by_route_and_id()
+  defp directions_by_route_id(directions_data) do
+    directions_data
+    |> Csv.parse(fn _row -> true end, &Direction.from_csv_row/1)
+    |> Enum.reduce(%{}, fn direction, acc ->
+      put_in(
+        acc,
+        Enum.map([direction.route_id, direction.direction_id], &Access.key(&1, %{})),
+        direction
+      )
+    end)
```

Comment:
Would [`&Enum.group_by/3`](https://hexdocs.pm/elixir/Enum.html#group_by/3) be useful here?

Reply:
Maybe I'm just missing the solution, but I don't think that `group_by` is able to do this—we are making a 2-level map instead of just a single one is the trouble. So the result of this is a map of type `%{Route.id() => %{Direction.id() => Direction.t()}}`

Reply:
Yeah, the multiple levels makes it more complicated. What about:
```elixir
|> Csv.parse(...)
|> Enum.group_by(fn direction -> direction.route_id end)
|> Helpers.map_values(fn directions_for_route ->
  Enum.group_by(directions_for_route, fn direction -> direction.direction_id end)
end)
```
How does that compare to doing a single reduce?

Reply:
I have to try to parse and understand this, but as a starting point what problem is it trying to solve or improve?

Reply:
Just avoiding having to write your own reduce. Usually you don't need reduce because there are so many existing functions built on top of it. In this case, the shape of the data is unusual enough that a custom reduce might be better.